internal/tasks/device_render.go

```diff
@@ -333,6 +374,13 @@ func (t *DeviceRenderLogic) renderGitConfig(ctx context.Context, configItem *api
 		}
 	}
 
+	// Validate all file paths produced by the git config before merging
+	for _, f := range ignition.Storage.Files {
+		if err := denyUnsafeDevicePath(f.Path); err != nil {
+			return &gitSpec.Name, &gitSpec.GitRef.Repository, fmt.Errorf("invalid path from git config: %w", err)
```

Comment:
note: I know you didnt setup this file but I don't understand why in error condition why we are not just returning nil for name and repo